sdlib/d/gc/arena.d

```diff
@@ -751,3 +815,101 @@ unittest resizeLarge {
 	// Now full again:
 	assert(pd9.extent.hpd.full);
 }
+
+unittest resizeLargeGrow {
+	import d.gc.arena;
+	shared Arena arena;
+
+	auto base = &arena.base;
+	scope(exit) arena.base.clear();
+
+	import d.gc.emap;
+	static shared ExtentMap emap;
+	emap.tree.base = base;
+
+	import d.gc.region;
+	shared RegionAllocator regionAllocator;
+	regionAllocator.base = base;
+
+	arena.regionAllocator = &regionAllocator;
+
+	PageDescriptor test_alloc_large(uint pages) {
+		auto ptr = arena.allocLarge(&emap, pages * PageSize);
+		assert(ptr !is null);
+		auto pd = emap.lookup(ptr);
+		assert(pd.extent.address is ptr);
+		assert(pd.extent.size == pages * PageSize);
+		return pd;
+	}
+
+	void test_grow_large(PageDescriptor pd, uint pages) {
+		assert(arena.resizeLarge(&emap, pd.extent, pages * PageSize));
+		assert(pd.extent.size == pages * PageSize);
+	}
+
+	// Allocation 0: 35 pages:
+	auto pd0 = test_alloc_large(35);
+
+	// Allocation 1: 64 pages:
+	auto pd1 = test_alloc_large(64);
+	assert(pd1.extent.hpd == pd0.extent.hpd);
```

Comment:
What you care the most about is that these allocation are contiguous, not really that they are on the same hpd (they will be if they are contiguous, and if they aren't, then you are not testing at all what you think you are testing).